x-pack/plugins/ml/public/application/datavisualizer/file_based/components/results_links/results_links.tsx

```diff
@@ -37,11 +39,37 @@ export const ResultsLinks: FC<Props> = ({
   });
   const [showCreateJobLink, setShowCreateJobLink] = useState(false);
   const [globalStateString, setGlobalStateString] = useState('');
+  const [globalState, setGlobalState] = useState<MlCommonGlobalState | undefined>();
+
   const {
     services: {
       http: { basePath },
     },
   } = useMlKibana();
+  const mlUrlGenerator = useMlUrlGenerator();
+  const navigateToPath = useNavigateToPath();
+
+  const openInDataVisualizer = useCallback(async () => {
+    const path = await mlUrlGenerator.createUrl({
+      page: ML_PAGES.DATA_VISUALIZER_INDEX_VIEWER,
+      pageState: {
+        index: indexPatternId,
+        globalState,
+      },
+    });
+    await navigateToPath(path);
+  }, [indexPatternId, globalState]);
+
+  const createNewMlJob = useCallback(async () => {
```

Comment:
I find this name misleading, it doesn't actually create a new ML job

Reply:
Renamed here https://github.com/elastic/kibana/pull/76735/commits/cea397ee78ebf873893a6bc05185ddacaa40704c